### ISSUE TITLE
How to use Med-NCA on a custom 2D dataset

### ISSUE THREAD

**Issue by alvinkimbowa** [MECLabTUDA/M3D-NCA#1]
Hi,

I have 2D ultrasound png images. How can I adapt the model to use my custom dataset loader?

My initial attempts have been to modify the load_item() function in the Nii_Gz_Dataset_3D as shown below;
```
def load_item(self, path):
      r"""Loads the data of an image of a given path.
          #Args
              path (String): The path to the nib file to be loaded."""
      # return nib.load(path).get_fdata()
      img = np.asarray(Image.open(path).resize((512, 512)))
      img = np.expand_dims(img, axis=-1)
      return img
```
I'm using the train_Med_NCA.ipynb notebook with the following parameters
```
config = [{
    'img_path': r"DatasetXXXX/imagesTr/",
    'label_path': r"DatasetXXXX/labelsTr/",
    'model_path': r'Models/Med_NCA_Run1',
    'device':"cuda:1",
    'unlock_CPU': True,
    # Optimizer
    'lr': 16e-4,
    'lr_gamma': 0.9999,
    'betas': (0.5, 0.5),
    # Training
    'save_interval': 2,
    'evaluate_interval': 2
    ,
    'n_epoch': 1000,
    'batch_size': 8,
    # Model
    'channel_n': 32,        # Number of CA state channels
    'inference_steps': 64,
    'cell_fire_rate': 0.5,
    'input_channels': 1,
    'output_channels': 1,
    'hidden_size': 128,
    'train_model':1,
    # Data
    'input_size': [(64, 64), (256, 256)],
    'data_split': [0.7, 0.15, 0.15], 
}
]
```
```
dataset = Dataset_NiiGz_3D(slice=2)
device = torch.device(config[0]['device'])
ca1 = BackboneNCA(config[0]['channel_n'], config[0]['cell_fire_rate'], device, hidden_size=config[0]['hidden_size'], input_channels=config[0]['input_channels']).to(device)
ca2 = BackboneNCA(config[0]['channel_n'], config[0]['cell_fire_rate'], device, hidden_size=config[0]['hidden_size'], input_channels=config[0]['input_channels']).to(device)
ca = [ca1, ca2]
agent = Agent_Med_NCA(ca)
exp = Experiment(config, dataset, ca, agent)
dataset.set_experiment(exp)
exp.set_model_state('train')
data_loader = torch.utils.data.DataLoader(dataset, shuffle=True, batch_size=exp.get_from_config('batch_size'))

loss_function = DiceBCELoss()
```

The model trains, but the DICE remains zero throughout the training and the loss remains flat. The image below is a sample prediction obtained from Tensorboard at the end of the training. Ideally, the cartilage should be segmented but it seems the model.

![individualImage](https://github.com/user-attachments/assets/77379e67-5e8c-4789-9c77-9d27b8d13936)

Note that the model works fine for the hippocampus and prostate datasets. Below is a sample prediction from the hippocampus dataset.

![image](https://github.com/user-attachments/assets/bf5a4e97-f172-4403-a87e-b257e7bc678a)

How can I adapt the model to my 2D dataset?

Thanks!

**Comment by alvinkimbowa** [comment]
Changing the learning rate to 1e-4 worked for me.
